### PR TITLE
adding support for stanag4676 format (issue #515)

### PR DESCRIPTION
**Pull request by rfecher**

### CODE REVIEW

File: extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/IngestMessageHandler.java [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+								BufferedImage.TYPE_3BYTE_BGR);
+						chipInfo.getValue().setImage(
+								scaledImage);
+						final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+						try {
+							ImageIO.write(
+									scaledImage,
+									DEFAULT_IMAGE_FORMAT,
+									baos);
+							chipInfo.getValue().setImageBytes(
```
Comment (dcy2003) [inline extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/IngestMessageHandler.java]: boas is flushed after toByteArray()

File: extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/Stanag4676IngestPlugin.java [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+
+					if (prevEv != null) {
+						distanceKm += prevEv.getDistance(currentEv);
+					}
+
+					// populate coordinate sequence
+					coord_sequence.add(event.Longitude.get());
+					coord_sequence.add(event.Latitude.get());
+
+					prevEv = new EarthVector(
```
Comment (dcy2003) [inline extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/Stanag4676IngestPlugin.java]: Can prevEv simply be set to currentEv?

**Comment by jwomeara** [comment]
Gasp!  No unit test?!

File: extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/IngestMessageHandler.java [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+		}
+	}
+
+	private static BufferedImage toBufferedImage(
+			final Image image,
+			final int type ) {
+		if (image instanceof BufferedImage) {
+			return (BufferedImage) image;
+		}
+		final BufferedImage bi = new BufferedImage(
```
Comment (dcy2003) [inline extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/IngestMessageHandler.java]: Use ImageChipUtils.toBufferedImage() ?

File: extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/parser/NATO4676Decoder.java [hunk 4]
```diff
@@ -0,0 +1,1182 @@
...
+	@Override
+	public TrackMessage readNext(
+			final InputStream is ) {
+		TrackMessage msg = null;
+		try {
+			if (printNotParse) {
+				final String trackStr = IOUtils.toString(
+						is,
+						"UTF-8");
+				System.out.println(trackStr);
```
Comment (dcy2003) [inline extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/parser/NATO4676Decoder.java]: Remove println() ?

**Pull request merged**